Create and fill the product attributes lookup table during a clean install of WooCommerce

Konamiman opened the pull request:
### All Submissions:

* [ ] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

https://github.com/woocommerce/woocommerce/pull/31256 was implemented with the wrong assumption that database migration functions are executed during a clean install of WooCommerce, but actually these functions are run only during an upgrade from a previous WooCommerce version. As a result, the table was not created during a clean install, and the feature wasn't visible at all. This pull request fixes this.

### How to test the changes in this Pull Request:

1. Build a WooCommerce .zip file from this branch, and use it to perform a clean install on a WordPress instance.
2. Verify that the tools page shows a "Regenerate the product attributes lookup table" entry.
3. Verify that Settings - Products has an "Advanced" section that allows controlling the feature.
4. Verify that the "Use the product attributes lookup table for catalog filtering" option in the above settings section is checked.
5. Verify that the above is true too when upgrading from a previous version of WooCommerce, in two cases: the lookup table had been already created via the dedicated tool, and the lookup table didn't exist previously. If the table already existed it should still be populated with the appropriate data, without needing to trigger a regeneration.

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry

> Fix - Product attributes lookup table not being created during a clean install of WooCommerce

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

Konamiman committed:
Create and fill the product attributes lookup table in clean installs.

The table creation and filling was happening in a data migration, but
data migrations only run when WooCommerce is upgraded, not on a
clean install. Also, the option for using the lookup table
wasn't being set after a clean install (when the table has been created
and is empty since there are no products yet).

As part of the fix the table is added to the list of WooCommerce
core tables, and thus it's covered by the "Verify base database tables"
tool.

peterfabian commented on plugins/woocommerce/includes/wc-update-functions.php:
```diff
@@ -2348,7 +2348,11 @@ function wc_update_630_create_product_attributes_lookup_table() {
+	 * If the table exists and contains data, it was manually created by user before the migration ran.
+	 * If the table exists but is empty, it was likely created right now via dbDelta, so a table regenerations is needed.
+	 */
+	if ( $data_store->check_lookup_table_exists() && $data_store->lookup_table_has_data() ) {
```
If we're not checking other edge cases like user deleting the table, then it seems unnecessary to check whether the table was added manually as well. Or should this handle stores that activated this feature in its previous version when it didn't include indices yet?

Konamiman replied:
Yes, exactly. This covers all the cases: table not existing, table existing with or without data, table existing with or without indices.

peterfabian replied:
Just for completeness, pasting it here as well:
it seems to me that dbDelta takes care of the indices

This is what I did:
1. Installed WC 6.2
2. Added products
3. Enabled the lookup table
4. Created a zip pkg from your branch but commented out `$data_regenerator->maybe_create_table_indices()` from `wc_update_630_create_product_attributes_lookup_table`;
5. Ran the update
6. Checked the indices were in place after the update

So unless I'm missing a scenario, `maybe_create_table_indices` method might be unnecessary?

peterfabian reviewed (dismissed):
Tested clean site, update from 6.2 without the feature enabled and update from 6.2 with the feature enabled. All ended up with a table full of data with correct indices, so seems to be working fine. 👍

Konamiman committed:
Remove the DataRegenerator::maybe_create_table_indices

It's not needed since dbDelta will take care of the indices in both
new WooCommerce installs and upgrades from a previous version.

peterfabian reviewed (approved):
Looking good now, thanks! Feel free to merge after the checks finish.
